src/test/run-pass/const-int-saturating-arith.rs

```diff
@@ -0,0 +1,27 @@
+const INT_U32_NO: u32 = (42 as u32).saturating_add(2);
+const INT_U32: u32 = u32::max_value().saturating_add(1);
+const INT_U128: u128 = u128::max_value().saturating_add(1);
+const INT_I128: i128 = i128::max_value().saturating_add(1);
+const INT_I128_NEG: i128 = i128::min_value().saturating_add(-1);
+
+const INT_U32_NO_SUB: u32 = (42 as u32).saturating_sub(2);
```

Comment:
It would be good to have saturating `INT_U32` and `INT_U128` test cases for sub as well. I think this is currently not handled correctly.